tutor/src/screens/assignment-edit/clone/editor.js

```diff
@@ -0,0 +1,162 @@
+import { React, PropTypes, styled, observer } from 'vendor';
+import { Modal, Button } from 'react-bootstrap';
+import { Formik } from 'formik';
+import UX from '../ux';
+import CloneModal from 'components/course-modal';
+import { colors, breakpoints } from 'theme';
+
+const StyledTemplateModal = styled(CloneModal)`
+  .modal-dialog {
+    margin-top: 3rem;
+    max-width: 90%;
+  }
+
+  .modal-body {
+    background: ${colors.white};
+    padding: 0 1rem;
+  }
+
+  &&& {
+    .modal-footer {
+      border-top: 2px solid ${colors.neutral.pale};
+      background-color: ${colors.neutral.bright};
+      padding: 1rem 2rem;
+      display: flex;
+      justify-content: space-between;
+
+      /* Cancel buttons */
+      & div:first-child {
+        .btn {
+          background: white;
+        }
+      }
+      /* Save as Draft, Publish, and Edit other details buttons */
+      & div:last-child {
+        .btn:not(:last-child) {
+          background: ${colors.neutral.lite};
+          border: 1px solid ${colors.neutral.lite};
+        }
+      }
+    }
+  }
+
+  @media ${breakpoints.xlUp} {
+  .modal-dialog {
+    max-width: 65%;
+  }
+}
+`;
+
+const WrqNote = styled.div`
+  padding: 0 1rem;
+  ul {
+    padding: 0 1.5rem;
+    color: ${colors.neutral.dark};
+  }
+`;
+
+@observer
+class Editor extends React.Component {
+
+  static propTypes = {
+    ux: PropTypes.instanceOf(UX).isRequired,
+    showModal: PropTypes.bool.isRequired,
+    onHide: PropTypes.func.isRequired,
+  }
+
+  async saveDraft() {
+    const { ux: { saveAsDraft }, onHide } = this.props;
+    await saveAsDraft();
+    onHide();
+  }
+
+  async publish() {
+    const { ux: { savePlan }, onHide } = this.props;
+    await savePlan();
+    onHide();
+  }
+
+  async editDetails() {
+    const { ux: { saveAsDraft, navigateToStep } } = this.props;
+    await saveAsDraft();
+    navigateToStep('details');
+  }
+
+  onCancel() {
+    this.props.onHide();
+  }
```

Comment:
functionally looks good, but these methods should be decorated with `@action` so that mobx only fires updates once.  Plus if you make them `@action.bound` you wouldn't need to use functions in the JSX callbacks below.